File: nim/grade-school/grade_school.nim
from algorithm import sortedByIt
from sequtils import mapIt, filterIt

type Student = tuple[name: string, grade: int]

proc sort(students: seq[Student]): seq[Student] =
    students.sortedByIt((it.grade, it.name))


type
    School* = object
        students*: seq[Student]

proc roster*(school: School): seq[string] =
    school.students.sort.mapIt(it.name)

proc grade*(school: School, grade: int): seq[string] =
    school.students.filterIt(it.grade == grade).sort.mapIt(it.name)

File: python/grade-school/grade_school.py
from collections import defaultdict


class School:
    def __init__(self):
        self.data = defaultdict(list)

    def add_student(self, name, grade):
        self.data[grade].append(name)

    def roster(self):
        return [name for k in sorted(self.data.keys()) for name in sorted(self.data[k])]

    def grade(self, grade_number):
        return sorted(self.data[grade_number])
